source/agora/registry/NameRegistryAPI.d

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+
+    Definitions of the name registry API
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.registry.NameRegistryAPI;
+
+import agora.common.Hash;
+import agora.common.Types;
+import agora.common.crypto.Key;
+
+import vibe.http.common;
+import vibe.web.rest;
+
+///
+struct RegistryPayloadData
+{
+    /// the public key that we want to register
+    public PublicKey public_key;
+
+    /// network addresses associated with the public key
+    public Address[] addresses;
+
+    /// monotonically increasing sequence number
+    public long seq;
```

Comment:
```suggestion
    public ulong seq;
```

Reply:
time() returns clock_t on POXIS and 
alias c_long     clock_t;

will this cause problems?

time actually could be negative if we want to refer to time before the unix epoch

Reply:
Then it's not a sequence number ?